src/storage/src/source/reclock.rs

```diff
@@ -559,12 +559,22 @@ where
                 upper = Antichain::new();
             }
 
+            // If this is the first binding we mint then we will mint it at the minimum target
+            // timestamp. The first source upper is always the upper of the snapshot and by mapping
```

Comment:
Is this guaranteed? The `remap_operator` can mint an empty source frontier, which will hit this block and map empty to empty, right?

Reply:
The `remap_operator` will not call `mint` until it receives a non-minimum progress frontier from the source, which should correspond to the snapshot

Reply:
I totally missed the changes on the `remap_operator` you made in this same pr; i was reading it locally!